AGENTS.md

```diff
@@ -516,6 +401,300 @@ export const MyResource = Resource(
 );
 ```
 
+### Advanced Resource Patterns
+
+#### Input Normalization with Wrapper Functions
+
+When resources accept flexible input types (e.g., `string | Secret`, `string | Resource`), use a public wrapper function to normalize inputs before passing to the internal Resource:
+
+```ts
```

Comment:
idk if we always want to do this, seems a little extreme to do it if its only 1 input type

Reply:
I can try to make it more clear when its NOT necessary.